lib/npm.js

```diff
@@ -322,9 +323,18 @@ NPMLocation.prototype = {
 
     var doAuth = this.alwaysAuth || repo[0] == '@';
 
+    var tarball = url.parse(versionData.dist.tarball);
```

Comment:
Can we include a comment here to note that this is from https://github.com/npm/npm/blob/50ce116baac8b6877434ace471104ec8587bab90/lib/cache/add-named.js#L196-L208